test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs

```diff
@@ -0,0 +1,320 @@
+using Test.Utility;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    [Collection(MockedVS.Collection)]
+    public class NuGetPackageSearchServiceTests : MockedVSCollectionTests
+    {
+        private readonly SourceRepository _sourceRepository;
+        private readonly IEnumerable<IPackageReferenceContextInfo> _installedPackages;
+        private readonly IReadOnlyCollection<IProjectContextInfo> _projects;
+
+        public NuGetPackageSearchServiceTests(GlobalServiceProvider globalServiceProvider)
+            : base(globalServiceProvider)
+        {
+            _installedPackages = new List<IPackageReferenceContextInfo>();
+            _projects = new List<IProjectContextInfo>
+            {
+                new ProjectContextInfo(
+                    Guid.NewGuid().ToString(),
+                    ProjectModel.ProjectStyle.PackageReference,
+                    NuGetProjectKind.PackageReference)
+            };
+            var testFeedUrl = "https://testsource.com/v3/index.json";
+            var query = "https://api-v2v3search-0.nuget.org/query";
+            var responses = new Dictionary<string, string>
+            {
+                { testFeedUrl, ProtocolUtility.GetResource("NuGet.PackageManagement.VisualStudio.Test.compiler.resources.index.json", GetType()) },
+                { query + "?q=nuget&skip=0&take=26&prerelease=true&semVerLevel=2.0.0", ProtocolUtility.GetResource("NuGet.PackageManagement.VisualStudio.Test.compiler.resources.nugetSearchPage1.json", GetType()) },
+                { query + "?q=nuget&skip=25&take=26&prerelease=true&semVerLevel=2.0.0", ProtocolUtility.GetResource("NuGet.PackageManagement.VisualStudio.Test.compiler.resources.nugetSearchPage2.json", GetType()) },
+                { query + "?q=&skip=0&take=26&prerelease=true&semVerLevel=2.0.0", ProtocolUtility.GetResource("NuGet.PackageManagement.VisualStudio.Test.compiler.resources.blankSearchPage.json", GetType()) },
+                { "https://api.nuget.org/v3/registration3-gz-semver2/nuget.core/index.json", ProtocolUtility.GetResource("NuGet.PackageManagement.VisualStudio.Test.compiler.resources.nugetCoreIndex.json", GetType()) },
+                { "https://api.nuget.org/v3/registration3-gz-semver2/microsoft.extensions.logging.abstractions/index.json", ProtocolUtility.GetResource("NuGet.PackageManagement.VisualStudio.Test.compiler.resources.loggingAbstractions.json", GetType()) }
+            };
+
+            _sourceRepository = StaticHttpHandler.CreateSource(testFeedUrl, Repository.Provider.GetCoreV3(), responses);
+        }
+
+        [Fact]
+        public async Task GetTotalCountAsync_WithGreaterThanOrEqualToMaxCountResults_ReturnsMaxCount()
+        {
+            var source1 = new PackageSource("https://dotnet.myget.org/F/nuget-volatile/api/v3/index.json", "NuGetVolatile");
+            var source2 = new PackageSource("https://api.nuget.org/v3/index.json", "NuGet.org");
```

Comment:
dotnet.myget.org is at risk of disappearing, making this test fail.

Plus, this is in a `.Test` project, not `.FuncTest`, so it's a unit test that should be fast (so `build.ps1` is fast), not making slow network requests. But as mentioned in another comment, our tests should not, in my opinion, be doing integration tests with different server implementations. We should use mock server for tests.

Reply:
tracking this problem with https://github.com/NuGet/Home/issues/10393